test/PositionToken.ts

```diff
@@ -1,15 +1,116 @@
+        const secondTokenId = 1;
+        const thirdTokenId = 2;
+        let firstTokenOwner: SignerWithAddress;
+        let firstTokenOwnerAddress: string;
+        let secondTokenOwner: SignerWithAddress;
+        let secondTokenOwnerAddress: string;
+        let thirdTokenOwner: SignerWithAddress;
+        let thirdTokenOwnerAddress: string;
+
+        before(async function () {
+            r = await buildContractTestContext();
+            await r.positionToken.init(r.owner.address);
+            firstTokenOwner = r.addr1;
+            firstTokenOwnerAddress = r.addr1.address;
+            secondTokenOwner = r.addr2;
+            secondTokenOwnerAddress = r.addr2.address;
+            thirdTokenOwner = r.addr3;
+            thirdTokenOwnerAddress = r.addr3.address;
+        });
+
+        it("Should revert if the token id doesn't exist", async function () {
+            await expect(r.positionToken.ownerOf(0)).to.be.reverted;
+        });
+
+        it("Should be mintable from address designated executive", async function () {
+            await r.positionToken.safeMint(r.addr1.address);
+            expect(await r.positionToken.ownerOf(firstTokenId)).to.equal(firstTokenOwnerAddress);
+        });
+
+        it("Should increment the positionTokenId properly", async function () {
+            await r.positionToken.safeMint(r.addr2.address);
+            expect(await r.positionToken.ownerOf(secondTokenId)).to.equal(secondTokenOwnerAddress);
+        });
+
+        it("Should not be mintable from an address other than LeverageEngine", async function () {
+            expect(r.positionToken).to.not.be.undefined;
+        });
+
+        it("Should fail to burn if not executive", async function () {
+            const burnPromise = r.positionToken.connect(secondTokenOwner).burn(firstTokenId);
+            await expect(burnPromise).to.be.revertedWith("onlyExecutive: Caller is not executive");
+        });
+
+        it("Should not allow positionToken owner to burn positionToken directly. Position unwind required via executive", async function () {
+            const burnPromise = r.positionToken.connect(firstTokenOwner).burn(firstTokenId);
+            await expect(burnPromise).to.be.revertedWith("onlyExecutive: Caller is not executive");
+        });
+
+        it("Should allow executive to burn any token", async function () {
+            await r.positionToken.burn(firstTokenId);
+            await r.positionToken.burn(secondTokenId);
+            /* should revert since token has been burned */
+            await expect(r.positionToken.burn(firstTokenId)).to.be.revertedWith("ERC721: operator query for nonexistent token");
+            await expect(r.positionToken.burn(secondTokenId)).to.be.revertedWith("ERC721: operator query for nonexistent token");
+        });
+
+        it("Should continue to increment id properly after tokens have been burned", async function () {
+            await r.positionToken.safeMint(r.addr3.address);
```

Comment:
returned NFT ID (from solidity) would be really helpful

Reply:
Yeah so this is what I wanted to test originally but due to transactions not returning the return value needed to check ownerOf below. However agreed it needs to return the id for use in contracts that call this and need it. Added here 6f0662117d6387314ffee6bf2b9d23301dbbb06a